src/R/Editor/Impl/Completion/Providers/FilesCompletionProvider.cs

```diff
@@ -50,14 +50,18 @@ enum Mode {
             _forceR = forceR;
 
             _enteredDirectory = ExtractDirectory(directoryCandidate);
-            if (_enteredDirectory.Length == 0 || _enteredDirectory.StartsWithOrdinal(".")) {
+            _task = GetRootDirectoryAsync(_enteredDirectory);
+        }
+
+        private Task<string> GetRootDirectoryAsync(string userProvidedDirectory) {
```

Comment:
There is a problem calling `t = asyncMethod()` in ctor. I spent some time yesterday on it. Basically task never runs and `.Wait` in `GetEntries` never completes (status == WaitingForActivation). As an experiment I removed Wait and just used `t.Result` which hangs. There is something with sync context. Perhaps this is b/c `ctor` and `GetEntries` are called on the same UI thread sequentially and task that is configured to return data to the main thread never completes?